### PR TITLE
minor fix for raft test

### PR DESCRIPTION
**Pull request by kikimo**
The msg type check and reject check at the end of `TestLearnerCanVote` is incorrect

### CODE REVIEW

**Review by ahrtr — APPROVED** [review]
LGTM

Thanks @kikimo

File: raft_test.go [hunk 1]
```diff
@@ -465,7 +465,7 @@ func TestLearnerCanVote(t *testing.T) {
 		t.Fatalf("expected exactly one message, not %+v", msgs)
 	}
 	msg := msgs[0]
-	if msg.Type != pb.MsgVoteResp && !msg.Reject {
+	if msg.Type != pb.MsgVoteResp || msg.Reject {
 		t.Fatal("expected learner to not reject vote")
 	}
```
Comment (pavelkalinnikov) [inline raft_test.go:470]: While here, could convert this to use the `require` package:

```golang
require.Len(t, msgs, 1)
require.Equal(t, msgs[0].Type, pb.MsgVoteResp)
require.False(t, msgs[0].Reject, "expected learner to not reject vote")
```

**Review by pavelkalinnikov — APPROVED** [review]

**Comment by ahrtr** [comment]
Could you squash the commits?

**Comment by kikimo** [comment]
> Could you squash the commits?

Squashed.

**Pull request merged**